Tools/Create-Azure-Sentinel-Solution/pipeline/createSolutionV4.ps1

```diff
@@ -140,12 +141,28 @@ try
 
 						try {
 							Write-Host "Downloading $finalPath"
-							$rawData = (New-Object System.Net.WebClient).DownloadString($finalPath)
+							$isFilePathPresent = Test-Path -Path "$finalPath"
+							Write-Host "Is $finalPath file path present $isFilePathPresent"
+							if ($isFilePathPresent) {
+								$rawData = (New-Object System.Net.WebClient).DownloadString($finalPath)
+							}
+							else {
+								if ($fileExtension -eq "json" -or $fileExtension -eq "JSON") {
+									Write-Host "FinalPath $finalPath not found!"
+									if ($fileExtension -eq "json") {
+										$finalPath = $finalPath.Replace(".json", ".JSON")
+									} else {
+										$finalPath = $finalPath.Replace(".JSON", ".json")
+									}
+									Write-Host "Updated FinalPath is $finalPath"
+									$rawData = (New-Object System.Net.WebClient).DownloadString($finalPath)
+								}
+							}
 						}
 						catch {
 							Write-Host "Failed to download $finalPath -- Please ensure that it exists in $([System.Uri]::EscapeUriString($basePath))" -ForegroundColor Red
 							Send-AppInsightsExceptionTelemetry -InstrumentationKey $instrumentationKey -Exception $_.Exception -CustomProperties @{ 'RunId'="$runId"; 'PullRequestNumber'= "$pullRequestNumber"; 'ErrorDetails'="CreateSolutionV4 : Error occured in catch block: $_"; 'EventName'="CreateSolutionV4" }
-							break;
+							exit 1;
```

Comment:
why do you need to exit the session?

Reply:
added exit to fail the job as we identified error.